salt/minion.py

```diff
@@ -920,6 +921,8 @@ def __init__(self, opts, context=None):
         # Late setup of the opts grains, so we can log from the grains module
         import salt.loader
 
+        # need sync of custom grains as may be used in pillar compilation
+        salt.utils.extmods.sync(opts, "grains")
```

Comment:
Does this result in slow down when running salt-call commands?

Also, I'm seeing this error when i run this code with salt-call 

```
(py-3.10)  ch3ll@megan-precision5550  ~/git/salt   fix_65027 ±  salt-call saltutil.sync_grains
/home/ch3ll/.pyenv/versions/py-3.10/bin/salt-call:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  __import__('pkg_resources').require('salt==3006.3+94.g87434b76ca7')
[ERROR   ] Failed to sync grains module: 'master_uri'
local:
```

Are you able to replicate the same issue?

Reply:
This error also happens with just a `state.apply` command: `salt-call state.apply test`

Reply:
Didn't see that when I ran locally, will try in the morning. but was running against an updated released version of salt, not the current branch installed.

Reply:
@Ch3LL I am unable to reproduce that from the current build off the PR , salt-3006.3+259.g95a6267167-x86_64-deb.zip
```
salt-call --local state.apply test
.
.
.
[DEBUG   ] The functions from module 'highstate' are being loaded by dir() on the loaded module
[DEBUG   ] LazyLoaded highstate.output
[TRACE   ] data = {'local': {'test_|-do nothing_|-do nothing_|-nop': {'name': 'do nothing', 'changes': {}, 'result': True, 'comment': 'Success!', '__sls__': 'test', '__run_num__': 0, 'start_time': '10:54:49.621780', 'duration': '0.862 ms', '__id__': 'do nothing'}}}
local:
----------
          ID: do nothing
    Function: test.nop
      Result: True
     Comment: Success!
     Started: 10:54:49.621780
    Duration: 0.862 ms
     Changes:   

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:   0.862 ms
root@tdeb11:/home/david/test_dev#
```
I am running this on a Debian 11 Intel, VirtualBox VM, what platform did you run the above command on ?
Tried it with and without the test parameter, and before / after clearing out the cache  ```salt-call --local saltutil.clear_cache```, and not seen any issue.
I am using a clean box, perhaps there was something lingering around on yours ?, would be interested to know if there was, so could adjust for it etc.

Reply:
Able to duplicate the issue now, with a local salt-master and the command: ```salt-call test.versions```. Note the problem does not occur with command:
```
root@tdeb11:/home/david/test_dev# salt-call test.version
[ERROR   ] Failed to sync grains module: 'master_uri'
local:
    3006.3+259.g95a6267167
root@tdeb11:/home/david/test_dev# salt-call --local test.version
local:
    3006.3+259.g95a6267167
root@tdeb11:/home/david/test_dev# salt td11 test.version
td11:
    3006.3+259.g95a6267167
root@tdeb11:/home/david/test_dev# 
```

Reply:
Got a chicken and egg problem with master_uri since it is not yet set when syncing up the grains early.
This is only with the remote client (of course) -> ReqChannel -> syncWrapper - AsyncReqChannel and no master_uri in opts yet.